src/App.jsx

```diff
@@ -80,12 +115,59 @@ function App() {
             value: "Overdue"
           }
         ],
-        onFilter: (value, todo) => todo.status === value
+        onFilter: (value, todo) => {
+          return todo.status === value
+        }
       }
     ]
 
+  function handleFilterVal(e) {
+    console.log("OnChange fires")
+    setFilterVal(e.target.value)
+  }
+  function handleSearch() {
+    if (!filterVal) {
+      setFilteredData(data)
+    } else {
+      setFilteredData(
+        (data || []).filter(todo =>
+          todo.title.toLowerCase()
+            .includes(filterVal.toLowerCase())
+        )
+      )
```

Comment:
What is stopping you here from doing the checks for all the todo properties (`"timeStamp", "dueDate", "tags", "status", "description", "title"`) ???
When you add more checks for the other properties you can expand the search to all todo's fields just as originally required.

Reply:
I just want to keep the search simple right now. And we both decided that for now we will only implement search by title.